### PR TITLE
feat(route): add 51read.org

### PR DESCRIPTION
**Pull request by lazwa34**
## Involved Issue / 该 PR 相关 Issue

Close #

## Example for the Proposed Route(s) / 路由地址示例

```routes
/51read/article/152685
```

## New RSS Route Checklist / 新 RSS 路由检查表
  
- [x] New Route / 新的路由
  - [x] Follows [Script Standard](https://docs.rsshub.app/joinus/advanced/script-standard) / 跟随 [路由规范](https://docs.rsshub.app/zh/joinus/advanced/script-standard)
- [ ] Anti-bot or rate limit / 反爬/频率限制
  - [ ] If yes, do your code reflect this sign? / 如果有, 是否有对应的措施?
- [ ] [Date and time](https://docs.rsshub.app/joinus/advanced/pub-date) / [日期和时间](https://docs.rsshub.app/zh/joinus/advanced/pub-date)
  - [ ] Parsed / 可以解析
  - [ ] Correct time zone / 时区正确
- [ ] New package added / 添加了新的包
- [ ] `Puppeteer`

### CODE REVIEW

File: lib/routes/51read/article.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    const pageLength = $chapter('.ml-page select')
+        .find('option')
+        .map((_, option) => option.attribs.value)
+        .toArray().length;
+
+    const item: DataItem[] = [];
+    if (pageLength > 1) {
+        item.push(...(await createItem(chapter, pageLength - 1)));
+    }
+    item.push(...(await createItem(chapter, pageLength)));
```
Comment (TonyRL) [inline lib/routes/51read/article.ts:52]: Don't do page turns. Providing content from the last page is enough

File: lib/routes/51read/article.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+            title: $('h1').text(),
+            description: $('.kb-cot').html() || '',
+            link: url,
+        };
+    }) as Promise<DataItem>;
+
+const get = async (url: string, encoding = 'UTF-8') => {
+    const response = await ofetch(url, { responseType: 'arrayBuffer' });
+    return new TextDecoder(encoding).decode(response);
+};
```
Comment (TonyRL) [inline lib/routes/51read/article.ts:90]: `UTF-8` encoding content does not require decoding

File: lib/routes/51read/article.ts (hunk 1 above)
Comment (lazwa34) [reply lib/routes/51read/article.ts]: It has been modified.

File: lib/routes/51read/article.ts [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    const chapter = `https://m.51read.org/zhangjiemulu/${id}`;
+    const $chapter = load(await get(chapter));
+
+    const pageLength = $chapter('.ml-page select')
+        .find('option')
+        .map((_, option) => option.attribs.value)
+        .toArray().length;
+
+    const item: DataItem[] = [];
+    item.push(...(await createItem(chapter, pageLength)));
```
Comment (TonyRL) [inline lib/routes/51read/article.ts:49]:
```suggestion
    const item: DataItem[] = await createItem(chapter, pageLength)
```

File: lib/routes/51read/article.ts [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+            title: $('h1').text(),
+            description: $('.kb-cot').html() || '',
+            link: url,
+        };
+    }) as Promise<DataItem>;
+
+const get = async (url: string) => {
+    const response = await ofetch(url, { responseType: 'arrayBuffer' });
+    return new TextDecoder().decode(response);
+};
```
Comment (TonyRL) [inline lib/routes/51read/article.ts:87]: A normal ofetch does work.

File: lib/routes/51read/article.ts [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+    };
+}
+
+const createItem = async (baseUrl: string, page: number) => {
+    const url = `${baseUrl}/${page}`;
+    const $latest = load(await ofetch(url));
+    const item = await Promise.all(
+        $latest('.kb-jp li>a')
+            .map((_, chapter) => buildItem(chapter.attribs.href))
+            .toArray()
```
Comment (TonyRL) [inline lib/routes/51read/article.ts:67]:
```suggestion
            .toArray()
            .toReversed()
```

**Pull request merged**